Add Develop/HADK page

vigejolla opened the pull request.

rainemak commented on Tools/Hardware_Adaptation_Development_Kit/README.md:
```diff
@@ -11,7 +11,7 @@ If you want to get Sailfish OS running on a particular device then you're doing
 It's about working with device bootloaders, configuring and building the Linux kernel, low level Android™ systems, low level linux systems, C, C++ and shellscript coding, debugging and working blind - all fun stuff!
 
-There is a lot more detail at [the current HADK site](https://sailfishos.org/develop/hadk/).
+There is a lot more detail at [the current HADK site](/Develop/HADK/).
```
Following pages have a link to hadk as well:

Develop/HW_Adaptation/Sailfish_X_Xperia_Android_10_Build_and_Flash/README.md
Develop/HW_Adaptation/Sailfish_X_Xperia_Android_6_Build_and_Flash/README.md
Develop/HW_Adaptation/Sailfish_X_Xperia_Android_8_Build_and_Flash/README.md
Develop/HW_Adaptation/Sailfish_X_Xperia_Android_9_Build_and_Flash/README.md
Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md

Could you please update those as well.

vigejolla replied:
> Develop/HW_Adaptation/Sailfish_X_Xperia_Android_10_Build_and_Flash/README.md Develop/HW_Adaptation/Sailfish_X_Xperia_Android_6_Build_and_Flash/README.md Develop/HW_Adaptation/Sailfish_X_Xperia_Android_8_Build_and_Flash/README.md Develop/HW_Adaptation/Sailfish_X_Xperia_Android_9_Build_and_Flash/README.md Develop/HW_Adaptation/Sailfish_X_Xperia_Android_11_Build_and_Flash/README.md
> 

Oh, those had links to `https://sailfishos.org/hadk` - notice the missing "develop". That's why my grep didn't find those. Yeah, I'll update.

sledges commented:
Awesome effort to bring the layout over, but shouldn't we change the top banner to a heading, to align style with the docs?

vigejolla commented:
> Awesome effort to bring the layout over, but shouldn't we change the top banner to a heading, to align style with the docs?

Sorry, I don't quite get what you are getting at. The top banner is a heading - it's just like the heading on top of every page, with the (ok, maybe not so) minor difference in style.

rainemak commented on _layouts/hadk.html:
```diff
@@ -0,0 +1,242 @@
+        <path d="M13 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V9z"></path><polyline points="13 2 13 9 20 9"></polyline>
+      </svg>
+    </symbol>
+  </svg>
```
This svg block could be shared between default.html and this and included here.

rainemak commented on _layouts/hadk.html:
```diff
@@ -0,0 +1,242 @@
+      {% endif %}
+
+      <div class="search-overlay"></div>
+    {% endif %}
```
Could this site search be shared between default as well?

rainemak commented on _layouts/hadk.html:
```diff
@@ -0,0 +1,242 @@
+        {% endif %}
+      {% endunless %}
+      <div id="main-content" class="main-content" role="main">
+        <section class="hadk-header-background">   
```
Only differences between this and default.html are from this line to 164. Could we split duplicates into own html snippets and include them in here and default.html to avoid duplicating them? Thinking out loud but there could be snippets like svgs, sidebar, main header, footer, etc.  What do you think?

vigejolla replied:
I think that's a good idea.

rainemak commented on assets/css/just-the-docs-sfos.scss:
```diff
@@ -46,6 +46,63 @@ a:not([class]) {
     line-height: 1.3;
   }
+
+  .hadk-header-background {
```
I wouldn't refer to hadk in these classes. Calling this as ".header-background" would make it be reusable elsewhere as well if so wanted.

rainemak commented:
Very nice effort indeed!

sledges commented:
> > Awesome effort to bring the layout over, but shouldn't we change the top banner to a heading, to align style with the docs?
> 
> Sorry, I don't quite get what you are getting at. The top banner is a heading - it's just like the heading on top of every page, with the (ok, maybe not so) minor difference in style.

Precisely what I meant:) Remove the styling of the grey-background banner. But I guess it's fine as it is too, because this makes the HADK page stand out, being the starting point for any porter.
And Raine is right, if we want to re-use this style for any other pages that deem so.

vigejolla commented:
refactored the pages so that the main layout htmls now contain mostly includes, and the minor differences between the layouts are controlled via include directive on main_content_wrap.html. I also removed the references to hadk on css class names.

rainemak reviewed (approved):
Very nice! Thanks @vigejolla
